caffe/src/caffe/layers/molgrid_data_layer.cpp

```diff
@@ -1077,6 +1077,44 @@ void MolGridDataLayer<Dtype>::setReceptor(const vector<atom>& receptor, const ve
   batch_info[0].setReceptor(rec);
 }
 
+template <typename Dtype>
+void MolGridDataLayer<Dtype>::setReceptorFlex(const vector<atom>& receptor, const vector<vec>& flexcoords) {
+  CHECK_GT(batch_info.size(), 0) << "Empty batch info in setReceptor";
+
+  vector<float3> cs; cs.reserve(receptor.size());
+  vector<float> types; types.reserve(receptor.size());
+  vector<float> radii; radii.reserve(receptor.size());
+
+  //receptor atoms
+  for (unsigned i = 0, n = receptor.size(); i < n; i++) {
+    const atom& a = receptor[i];
+    smt origt = a.sm;
+    auto t_r = recTypes->get_int_type(origt);
+    int t = t_r.first;
+
+    vec coord;
+    if(i < flexcoords.size()){
+      // Get flex and inflex coords explicitly
+      coord = flexcoords[i];
```

Comment:
For flex and inflex atoms, coordinates are taken from `model.coords`, containing the coordinates of the movable atoms.

Reply:
Now removed because of the unified `setReceptor` function.